runtime/v2/runc/v2/service.go

```diff
@@ -21,61 +21,71 @@ package v2
 
 import (
 	"context"
-	"encoding/json"
-	"io/ioutil"
 	"os"
-	"path/filepath"
 	"sync"
-	"syscall"
-	"time"
 
 	"github.com/containerd/cgroups"
 	cgroupsv2 "github.com/containerd/cgroups/v2"
 	eventstypes "github.com/containerd/containerd/api/events"
 	"github.com/containerd/containerd/api/types/task"
 	"github.com/containerd/containerd/errdefs"
-	"github.com/containerd/containerd/mount"
 	"github.com/containerd/containerd/namespaces"
 	"github.com/containerd/containerd/pkg/oom"
 	oomv1 "github.com/containerd/containerd/pkg/oom/v1"
 	oomv2 "github.com/containerd/containerd/pkg/oom/v2"
 	"github.com/containerd/containerd/pkg/process"
+	"github.com/containerd/containerd/pkg/shutdown"
 	"github.com/containerd/containerd/pkg/stdio"
 	"github.com/containerd/containerd/pkg/userns"
+	"github.com/containerd/containerd/plugin"
 	"github.com/containerd/containerd/runtime/v2/runc"
 	"github.com/containerd/containerd/runtime/v2/runc/options"
+	runcservice "github.com/containerd/containerd/runtime/v2/runc/service"
 	"github.com/containerd/containerd/runtime/v2/shim"
+	shimapi "github.com/containerd/containerd/runtime/v2/task"
 	taskAPI "github.com/containerd/containerd/runtime/v2/task"
 	"github.com/containerd/containerd/sys/reaper"
 	runcC "github.com/containerd/go-runc"
+	"github.com/containerd/ttrpc"
 	"github.com/containerd/typeurl"
-	"github.com/gogo/protobuf/proto"
 	ptypes "github.com/gogo/protobuf/types"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	exec "golang.org/x/sys/execabs"
-	"golang.org/x/sys/unix"
 )
 
 var (
 	_     = (taskAPI.TaskService)(&service{})
 	empty = &ptypes.Empty{}
 )
 
-// group labels specifies how the shim groups services.
-// currently supports a runc.v2 specific .group label and the
-// standard k8s pod label.  Order matters in this list
-var groupLabels = []string{
-	"io.containerd.runc.v2.group",
-	"io.kubernetes.cri.sandbox-id",
-}
+// New returns a new shim service that can be used via GRPC
+// TODO(2.0): Remove this function, rely on plugin registration
+func New(_ context.Context, id string, _ shim.Publisher, _ func()) (shim.Shim, error) {
```

Comment:
Just glancing over the changes, and I'm a bit confused; is this backward compatible (as in; all arguments are now ignored, whereas previously they did something).

If the change is non-backward compatible from that perspective (even without the signature changing), I'm wondering if instead we should explicitly break it, and remove the arguments?

It doesn't look to be widely used (https://grep.app/search?q=github.com/containerd/containerd/runtime/v2/runc/v2), so mostly commenting (wondering if we want to stick with SemVer for these), but I see at least firecracker is using this; (cc @kzys) https://github.com/firecracker-microvm/firecracker-containerd/blob/568c84051caf61c1fc77a0b62171c22c40219b6a/agent/service.go#L99

Reply:
Not compatible for that firecracker use case. I opened this PR earlier so I can test that the shutdown logic is working. I can probably leave that function compatible though and just avoid it entirely for our own compilation. When we use it in the future, it can just register the task service. The update strips the task service from the "Shim", just leaving the shim manager components. The linked to use relies on the task service though. There is no real way to keep the interfaces clean and separate shim manager and task service though. The "2.0" comment will give a chance to completely clean that up though.

Reply:
> Not compatible for that firecracker use case. I opened this PR earlier so I can test that the shutdown logic is working. I can probably leave that function compatible though and just avoid it entirely for our own compilation. 

Thanks for clarifying; yes if we could keep it compatible (at least for now), that would be great